### DISCUSSION TITLE
ANTLR Kotlin fork

### DISCUSSION THREAD

**Discussion by Fancryer** [antlr/antlr4#4524] [Ideas]
Hello, I use ANTLR almost all the time when developing in Java/Kotlin, but I am a little frustrated that it is implemented exclusively in Java and uses quite a lot of deprecated methods and classes, as well as unsafe solutions. Can I fork it in Kotlin? I plan to implement null safety wherever possible, as well as keep the code cleaner and more functional. Yes, I understand that this is a big project, but the experience would be quite useful, I think. If necessary, I may limit its use to personal use.

**Comment by ericvergnaud** [comment]
Hi there,
you might want to explore the existing Kotlin implementation, see https://github.com/Strumenta/antlr-kotlin

**Comment by KvanTTT** [comment]
Also, we work on [ANTLR5](https://github.com/antlr/antlr5) that rely on Kotlin as well.

**Comment by SomeTroglodyte** [comment]
> might want to explore the existing Kotlin implementation

A bit too Gradle-heavy. I'd like to see a simple `antlr4 -Dlanguage=Kotlin -no-listener -visitor mygrammar.g4` from the command line succeed, and for that I hit the python/classpath barrier. How do I call antlr as java when the recommendation is to `pip install` it??? The python proxy file shows no hint on how to pass additional java arguments...
